docs/build-notifications.rst

```diff
@@ -20,31 +20,256 @@ Take these steps to enable build notifications using email:
+"Recent Activity" table. By clicking on each individual entry,
+you will see the server response, the webhook request, and the payload.
+
+.. figure:: /_static/images/webhooks-activity.png
+   :align: center
+   :alt: Activity of a webhook
+
+   Activity of a webhook
+
+Custom payload examples
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can customize the payload of the webhook to suit your needs,
+as long as it is valid JSON. Below you have a couple of examples,
+and in the following section you will find all the available variables.
+
+.. figure:: /_static/images/webhooks-payload.png
+   :width: 80%
+   :align: center
+   :alt: Custom payload
 
-The project name, slug and its details for the build will be sent as POST request to your webhook URL:
+   Custom payload
+
+Slack
++++++
+
+.. code-block:: json
+
+   {
+     "attachments": [
+       {
+         "color": "#db3238",
+         "blocks": [
+           {
+             "type": "section",
+             "text": {
+               "type": "mrkdwn",
+               "text": "*Read the Docs build failed*"
+             }
+           },
+           {
+             "type": "section",
+             "fields": [
+               {
+                 "type": "mrkdwn",
+                 "text": "*Project*: <{{project.url}}|{{project.name}}>"
+               },
+               {
+                 "type": "mrkdwn",
+                 "text": "*Version*: {{version.name}} ({{build.commit}})"
+               },
+               {
+                 "type": "mrkdwn",
+                 "text": "*Build*: <{{build.url}}|{{build.id}}>"
+               }
+             ]
+           }
+         ]
+       }
+     ]
+   }
+
+More information on `the Slack Incoming Webhooks documentation <https://api.slack.com/messaging/webhooks>`_.
+
+Discord
++++++++
 
 .. code-block:: json
 
+   {
+     "username": "Read the Docs",
+     "content": "Read the Docs build failed",
+     "embeds": [
        {
-            "name": "Read the Docs",
-            "slug": "rtd",
-            "build": {
-                "id": 6321373,
-                "commit": "e8dd17a3f1627dd206d721e4be08ae6766fda40",
-                "state": "finished",
-                "success": false,
-                "date": "2017-02-15 20:35:54"
-            }
+         "title": "Build logs",
+         "url": "{{build.url}}",
+         "color": 15258703,
+         "fields": [
+           {
+             "name": "*Project*",
+             "value": "{{project.url}}",
+             "inline": true
+           },
+           {
+             "name": "*Version*",
+             "value": "{{version.name}} ({{build.commit}})",
+             "inline": true
+           },
+           {
+             "name": "*Build*",
+             "value": "{{build.url}}"
+           }
+         ]
+       }
+     ]
+   }
+
+More information on `the Discord webhooks documentation <https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks>`_.
```

Comment:
I think we should maybe start each section with the links to the docs to create the webhhok on slack/discord and then give them the payload to use (Follow this steps on discord/slack, use the given URL when creating a webhook on rtd and add this payload).